### PR TITLE
[Tests] clean up and refactor gradient checkpointing tests

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Gradient checkpointing is an essential component of model training. We need to ensure it's implemented properly. 

If we had them implemented properly we could have avoided the fix from https://github.com/huggingface/diffusers/pull/9489 beforehand. Another related issue: https://github.com/huggingface/diffusers/issues/9496. 

Additionally, I took the liberty of properly skipping the related tests with "unittest.skip". This way, we know that tests are being actually skipped.

### CODE REVIEW

File: tests/models/autoencoders/test_models_vae.py [hunk 1]
```diff
@@ -364,9 +330,20 @@ def prepare_init_args_and_inputs_for_common(self):
...
     def test_outputs_equivalence(self):
         pass
 
+    def test_gradient_checkpointing_is_applied(self):
+        expected_set = {"DecoderTiny", "EncoderTiny"}
+        super().test_gradient_checkpointing_is_applied(expected_set=expected_set)
+
+    @unittest.skip(
+        "Gradient checkpointing is supported but this test doesn't apply to this class because it's forward is a bit different from the rest."
+    )
```
Comment (sayakpaul) [inline tests/models/autoencoders/test_models_vae.py:343]: Other forwards don't apply scaling and unscaling like it does. So, that makes this a little different.

**Comment by sayakpaul** [comment]
@yiyixuxu WDYT?

**Comment by sayakpaul** [comment]
@DN6 could you give this a look?

**Review by DN6 — APPROVED** [review]

**Comment by sayakpaul** [comment]
Failing test is unrelated.

**Pull request merged**